cpp/src/strings/convert/convert_durations.cu

```diff
@@ -186,11 +186,14 @@ __device__ void dissect_duration(T duration, duration_component* timeparts)
   }
 }
 
+namespace {
 template <typename T>
-struct duration_to_string_size_fn {
-  const column_device_view d_durations;
-  const format_item* d_format_items;
+struct from_durations_fn {
+  column_device_view d_durations;
+  format_item const* d_format_items;
   size_type items_count;
```

Comment:
Why none of these variables `const`? They won't change at all.

Reply:
They are data members, and marking data members as const impacts class' special members. IIRC, it disables the move ctor/assignment, because move operations would modify the "source" object. AFAIK we avoid const data members in libcudf for this reason.

Reply:
I suspect that too. But from my view I don't see anywhere the functor can be moved. Unless you explicitly use `std::move()` the functor is most likely to be copied.

Reply:
I'm glad for this discussion:
https://quuxplusone.github.io/blog/2022/01/23/dont-const-all-the-things/#data-members-never-const
On the face of it, it sounds like a bad idea.